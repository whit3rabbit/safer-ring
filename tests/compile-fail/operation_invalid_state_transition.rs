// This test should fail to compile because state transitions must follow the correct order

use safer_ring::{Operation};

fn main() {
    let operation = Operation::read()
        .fd(0);
    
    // This should not compile - cannot directly create a completed operation from building
    let _completed: Operation<'_, '_, safer_ring::operation::Completed<i32>> = operation;
}