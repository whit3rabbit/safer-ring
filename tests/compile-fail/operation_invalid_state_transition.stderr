error[E0308]: mismatched types
  --> tests/compile-fail/operation_invalid_state_transition.rs:10:80
   |
10 |     let _completed: Operation<'_, '_, safer_ring::operation::Completed<i32>> = operation;
   |                     --------------------------------------------------------   ^^^^^^^^^ expected `Operation<'_, '_, Completed<i32>>`, found `Operation<'_, '_, Building>`
   |                     |
   |                     expected due to this
   |
   = note: expected struct `Operation<'_, '_, Completed<i32>>`
              found struct `Operation<'_, '_, Building>`
